packages/edit-site/src/components/header-edit-mode/index.js

```diff
@@ -191,7 +191,7 @@ export default function HeaderEditMode( { setListViewToggleElement } ) {
 		>
 			{ hasDefaultEditorCanvasView && (
 				<NavigableToolbar
-					as={ motion.div }
+					render={ <motion.div /> }
```

Comment:
I don't know if the `as` prop was part of the public API of the `NavigableToolbar` component. But if it was, we should probably re-implement that within `NavigableToolbar`.

Something like this may work:

```jsx
const Component = props.as;
const render = Component ? <Component /> : undefined;
<Toolbar render={ render } />
```

Reply:
Thanks for the review!

It appears that the `as` prop in the `NavigableToolbar` component is [not defined](https://github.com/WordPress/gutenberg/blob/e373902346e4be9ca2b26595f15d0409cef0967d/packages/block-editor/src/components/navigable-toolbar/index.js#L157-L162) and is simply received as a remainder of the defined props. In this case, I thought it would be better not to re-implement internally from `as` prop to `render` prop, but what do you think?

Reply:
There's a few things happening here:

- `Toolbar` technically never supported the `as` prop `: the `false` [here](https://github.com/WordPress/gutenberg/blob/d84da61d9b35da1329ac70d9ac7d158862813160/packages/components/src/toolbar/toolbar/index.tsx#L28) in fact marks the component as non polymorphic
- But since the component internally used to spread the props to the underlying reakit component, passing the `as` prop worked (even if technically not supported)
- `Toolbar` was recently refactored from `reakit` to `ariakit`, which introduced the deprecation warning for the `as` prop

So, technically `Toolbar` doesn't support neither `as` nor `render`. On top of that, looking at the implementation of `NavigableToolbar`, depending on the `isAccessibleToolbar` prop, it could also render a `NavigableMenu`, which would ignore the `as` prop completely.

It just seems to me that the best choice here is to find a different way to animate the header without passing the `as` prop (or `render` prop) to `Toolbar`

Reply:
Maybe the other way around?
```jsx
const MotionNavigableToolbar = motion( NavigableToolbar );
// ...
<MotionNavigableToolbar>
```

Ref: https://www.framer.com/motion/component/#custom-components

Reply:
yeah we could try that

Reply:
Thank you for your ideas. I have tried various implementations based on the [documentation](https://www.framer.com/motion/component/#custom-components), but so far nothing has worked. The editor crashes with an error logged in the browser console, or the animation doesn't work.

```javascript
const MotionNavigableToolbar = motion( NavigableToolbar );
// ...
<MotionNavigableToolbar
 ...
/>
```
---
```javascript
const MotionNavigableToolbar = motion( 'NavigableToolbar' );
// ...
<MotionNavigableToolbar
 ...
/>
```
---
```javascript
const MotionNavigableToolbar = motion( 'navigable-toolbar' );
// ...
<MotionNavigableToolbar
 ...
/>
```
---
```javascript
const Component = forwardRef( ( props, ref ) => (
	<NavigableToolbar ref={ ref } { ...props } />
) );
const MotionNavigableToolbar = motion( Component );
// ...
<MotionNavigableToolbar
 ...
/>
```
---
```javascript
const Component = forwardRef( ( { children, ...props }, ref ) => (
	<NavigableToolbar { ...props } ref={ ref }>
		{ children }
	</NavigableToolbar>
) );
const MotionNavigableToolbar = motion( Component );
// ...
<MotionNavigableToolbar
 ...
/>
```

I'll try to find another approach.

Reply:
How about wrapping `NavigableToolbar` in a `motion.div` ?

Or simply adding the animation via SCSS ?

Reply:
> How about wrapping `NavigableToolbar` in a `motion.div` ?

In the end, I did this approach. The layout and animations should remain the same.